cardano-cli/test/cardano-cli-golden/Test/Golden/Help.hs

```diff
@@ -63,7 +63,47 @@ hprop_golden_HelpAll =
 
       help <-
         filterAnsi
-          <$> execCardanoCLI
+          <$> execCardanoCLIWithEnvVars
+            [ ("CARDANO_CLI_VISIBILITY_LEVEL", "visible")
+            ]
+            [ "help"
+            ]
+
+      H.diffVsGoldenFile help helpFp
+
+hprop_golden_HelpAllHidden :: Property
+hprop_golden_HelpAllHidden =
+  propertyOnce . H.moduleWorkspace "help" $ \_ -> do
+    -- These tests are not run on Windows because the cardano-cli usage
+    -- output is slightly different on Windows.  For example it uses
+    -- "cardano-cli.exe" instead of "cardano-cli".
+    unless isWin32 $ do
+      helpFp <- H.note "test/cardano-cli-golden/files/golden/help-hidden.cli"
+
+      help <-
+        filterAnsi
+          <$> execCardanoCLIWithEnvVars
+            [ ("CARDANO_CLI_VISIBILITY_LEVEL", "hidden")
+            ]
+            [ "help"
+            ]
+
+      H.diffVsGoldenFile help helpFp
+
+hprop_golden_HelpAllInternal :: Property
+hprop_golden_HelpAllInternal =
+  propertyOnce . H.moduleWorkspace "help" $ \_ -> do
+    -- These tests are not run on Windows because the cardano-cli usage
+    -- output is slightly different on Windows.  For example it uses
+    -- "cardano-cli.exe" instead of "cardano-cli".
+    unless isWin32 $ do
+      helpFp <- H.note "test/cardano-cli-golden/files/golden/help-internal.cli"
+
+      help <-
+        filterAnsi
+          <$> execCardanoCLIWithEnvVars
+            [ ("CARDANO_CLI_VISIBILITY_LEVEL", "internal")
```

Comment:
It's used here.

We ensure that `hidden` and `internal` cli commands are still tracked by golden files so we don't break them accidentally.